### PR TITLE
automatic hall of fame population

### PR DESCRIPTION
**Pull request by jgesel**
this PR automatically populates the HoF page. It does so by aggregating the OrderItems table, grouping by IceCreamID and summing quantities and then getting IceCream info foe each ID and returning the info for the 100 best selling flavors. This probably could all be done in one in mongoose statement but it was giving me some trouble so..... It also adds dependencies CORs dependencies as in this PR (https://github.com/larkinds/recurse-randomice/pull/70)
Solves #60

### CODE REVIEW

File: client/src/pages/hallOfFame/HallOfFame.tsx [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+import axios from "axios";
 
-const flavors: Flavor[] = Array.from({ length: 22 }, () => ({
-  name: "Vanilla",
-  price: 0,
-  image: "/strawberry-icecream.jpg",
-}));
+const hofFlavors = await axios
+  .get("http://localhost:3003/api/hof")
+  .then((result) => {
```
Comment (larkinds) [inline client/src/pages/hallOfFame/HallOfFame.tsx]: This .then is unnecessary - the result will be put into hofFlavors. Then below, you can do hofFlavors.data.forEach. Or, you can do it like this:

const response = await axios.get("url")
let hofFlavors = response.data

Checkout the section on await here:
https://javascript.info/async-await

**Review by larkinds — CHANGES_REQUESTED** [review]
Great working with setting up the hall of fame backend! Just one clean up item related to async/await statements.

File: client/src/pages/hallOfFame/HallOfFame.tsx [hunk 2]
```diff
@@ -2,14 +2,10 @@ import FlavorThumbnail, { Flavor } from "../../components/FlavorThumbnail";
...
-  .get("http://localhost:3003/api/hof")
-  .then((result) => {
-    return result.data;
-  });
+const hofFlavors = await axios.get("http://localhost:3003/api/hof");
 
 const flavors: Flavor[] = [];
-hofFlavors.forEach((flavor: any) => {
+hofFlavors.data.forEach((flavor: any) => {
   console.log(flavor);
```
Comment (larkinds) [inline client/src/pages/hallOfFame/HallOfFame.tsx]: clean up?

**Review by larkinds — APPROVED** [review]
Looks good, thanks for the refactor!

File: client/src/pages/hallOfFame/HallOfFame.tsx (hunk 2 above)
Comment (eyeriss42) [reply client/src/pages/hallOfFame/HallOfFame.tsx]: removed, thanks

File: client/src/pages/hallOfFame/HallOfFame.tsx (hunk 1 above)
Comment (eyeriss42) [reply client/src/pages/hallOfFame/HallOfFame.tsx]: Josh removed it, thanks!

**Pull request merged**